core/queryparser/sparql/src/main/java/org/eclipse/rdf4j/query/parser/sparql/SPARQLParser.java

```diff
@@ -57,6 +60,29 @@
 
 @SuppressWarnings("deprecation")
 public class SPARQLParser implements QueryParser {
+	private final Map<String, String> customPrefixes;
+
+	/**
+	 * Create a new SPARQLParser.
+	 * 
+	 * @param customPrefixes the default namespaces to apply to this parser.
+	 * @throws java.lang.NullPointerException if defaultPrefixes is null.
+	 */
+	public SPARQLParser(Set<Namespace> customPrefixes) {
+		Objects.requireNonNull(customPrefixes, "defaultPrefixes can't be null!");
+		if (customPrefixes.isEmpty()) {
+			this.customPrefixes = null;
```

Comment:
```suggestion
			this.customPrefixes = Collections.emptyMap();
```
An empty map would work well here.